demo/src/examples/load-media-form-component.js

```diff
@@ -0,0 +1,151 @@
+import { html, LitElement, unsafeCSS } from 'lit';
+import { animations, theme } from '../theme/theme';
+import componentCSS from 'bundle-text:./load-media-form-component.scss';
+import { classMap } from 'lit/directives/class-map.js';
+
+/**
+ * LoadMediaFormComponent is a LitElement that provides a user interface for loading media content.
+ *
+ * @element load-media-form
+ *
+ * @fires LoadMediaFormComponent#submit-media - Dispatched when the user submits media with the specified details.
+ *
+ * @prop {String} src - The URL or URN of the media content to be loaded.
+ * @prop {{vendor: String, certificateUrl: String, licenseUrl: String}} drmSettings - DRM settings for the loaded media.
+ */
+export class LoadMediaFormComponent extends LitElement {
+  static properties = {
+    src: { type: String },
+    drmSettings: { type: Object },
+    drmSettingsShown: { state: true, type: Boolean }
+  };
+
+  static styles = [theme, animations, unsafeCSS(componentCSS)];
+
+  constructor() {
+    super();
+
+    this.src = '';
+    this.#initDrmSettings();
+  }
+
+  #initDrmSettings() {
+    this.drmSettings = {
+      vendor: '',
+      certificateUrl: '',
+      licenseUrl: ''
+    };
+  }
+
+  #submitMedia() {
+    const src = this.src;
+    const type = src.startsWith('urn:') ? 'srgssr/urn' : undefined;
+    const keySystems = this.#keySystems;
+
+    /**
+     * Custom event dispatched by LoadMediaFormComponent when the user submits media.
+     *
+     * @event LoadMediaFormComponent#submit-media
+     * @type {CustomEvent}
+     * @property {Object} detail - The event detail object.
+     * @property {string} detail.src - The URL or URN of the media content to be loaded.
+     * @property {string | undefined} detail.type - The type of media. Undefined if the type cannot be determined.
+     * @property {Object | undefined} detail.keySystems - DRM key systems for the loaded media.
+     */
+    this.dispatchEvent(new CustomEvent('submit-media', {
+      detail: { src, type, keySystems }
+    }));
+  }
+
+  #handleLoadBarKeyUp(event) {
+    this.src = event.target.value;
+
+    if (event.key === 'Enter' && this.src) {
+      this.#submitMedia();
+    }
+  }
+
+  get #keySystems() {
+    if (!this.drmSettings?.vendor) {
+      return undefined;
+    }
+
+    const certificateUrl = this.drmSettings.certificateUrl;
+    const licenseUrl = this.drmSettings.licenseUrl;
+
+    if ('com.apple.fps.1_0' === this.drmSettings.vendor) {
+      return { [this.drmSettings.vendor]: { certificateUrl, licenseUrl }};
+    } else {
+      return { [this.drmSettings.vendor]: { licenseUrl }};
+    }
```

Comment:
Is it ok for you if we remove the `else` statement? I was never a fan of it 😅

Reply:
Done.